apps/api/src/modules/webhook/webhook.service.ts

```diff
@@ -0,0 +1,82 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { Webhook } from './entities/webhook.entity';
+import { Notification } from '../notifications/entities/notification.entity';
+import axios from 'axios';
+import { CreateWebhookInput } from './dto/create-webhook.input';
+import { Status } from 'src/common/constants/database';
+
+@Injectable()
+export class WebhookService {
+  protected readonly logger = new Logger(WebhookService.name);
+
+  constructor(
+    @InjectRepository(Webhook)
+    private readonly webhookRepository: Repository<Webhook>,
+  ) {}
+
+  async registerWebhook(webhookInput: CreateWebhookInput): Promise<Webhook> {
+    // Check if there is an active webhook with the same provider_id
+    const existingWebhook = await this.webhookRepository.findOne({
+      where: {
+        providerId: webhookInput.providerId,
+        status: Status.ACTIVE,
+      },
+    });
+
+    if (existingWebhook) {
+      throw new Error('This provider webhook exist');
+    }
+
+    const webhook = this.webhookRepository.create(webhookInput);
+    return await this.webhookRepository.save(webhook);
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
Can remove the comment.

Reply:
@Harish-osmosys  Update description and postman with Mutation.

Reply:
Updated